### ISSUE TITLE
Performance WPF

### ISSUE THREAD

**Issue by kyooi93** [oxyplot/oxyplot#1673]
### Steps to reproduce

1. Create ScatterSeries Graph
2. Generate around 5000 to 10000 point

Platform:
.NET version:

### Expected behaviour

1. Zoom in and out with smooth behavior
2. Auto size the graph 
### Actual behaviour

1 .when zoom in and out its hang and laggy. make whole program slow and hang.
2. Not resize the graph when plotting

![image](https://user-images.githubusercontent.com/71051053/94643255-f6399780-0318-11eb-8177-a7ea5326d571.png)

**Comment by VisualMelon** [comment]
#1609 should help with the performance side. Marker performance is known to be less than fantastic in WPF (and Avalonia).

What do you mean about it not resizing the plot?

**Comment by kyooi93** [comment]
> #1609 should help with the performance side. Marker performance is known to be less than fantastic in WPF (and Avalonia).
> 
> What do you mean about it not resizing the plot?

**Comment by kyooi93** [comment]
> #1609 should help with the performance side. Marker performance is known to be less than fantastic in WPF (and Avalonia).
> 
> What do you mean about it not resizing the plot?

**Comment by kyooi93** [comment]
> > #1609 should help with the performance side. Marker performance is known to be less than fantastic in WPF (and Avalonia).
> > What do you mean about it not resizing the plot?

The resize mean when the graph was plotted with data , the graph cant be auto zoom out to view the graph as overview.
Let me show you picture, will be more clearer.

**After Plot with Data**
![image](https://user-images.githubusercontent.com/71051053/94651429-05294580-032b-11eb-9721-c490fba0571a.png)

**Have to zoom out manually** 
![image](https://user-images.githubusercontent.com/71051053/94651482-18d4ac00-032b-11eb-91e3-6ac03b97a393.png)

**Comment by kyooi93** [comment]
If want to plot the graph like this what method can use to improve the performance ?

**Comment by VisualMelon** [comment]
Ah, right. You need to explicitly ask the plot to reset the view with a call to `PlotModel.ResetAllAxes();`

With regards to improving performance, I'm afraid I can't suggest anything. I'm not overly familiar with the WPF renderer. There is a new immediate-mode renderer based on Skia Sharp (which is generally much snappier), but that's only available in pre-release CI packages at the moment.

**Comment by kyooi93** [comment]
> Ah, right. You may need to explicitly ask the plot to reset the view with a call to `PlotModel.ResetAllAxes();`
> 
> With regards to improving performance, I'm afraid I can't suggest anything. I'm not overly familiar with the WPF renderer. There is a new immediate-mode renderer based on Skia Sharp (which is generally much snappier), but that's only available in pre-release CI packages at the moment.

![image](https://user-images.githubusercontent.com/71051053/94655047-918a3700-0330-11eb-9396-bec68849145b.png)

I am calling this function but still the same.

**Comment by VisualMelon** [comment]
Sorry, I misunderstood. You'll need to call `PlotModel.InvalidatePlot(true)` after changing any data (are you calling `PlotModel.InvalidatePlot(false)`?). `ResetAllAxes` needs to be called to reset the axes after any interaction if you want it to zoom the axes automatically. Could you provide a complete example?

**Comment by kyooi93** [comment]
> Sorry, I misunderstood. You'll need to call `PlotModel.InvalidatePlot(true)` after changing any data (are you calling `PlotModel.InvalidatePlot(false)`?). `ResetAllAxes` needs to be called to reset the axes after any interaction if you want it to zoom the axes automatically. Could you provide a complete example?

Yeah , The PlotModel.InvalidatePlot(true) would help me resize the graph after changing data. Sorry, I am not so familiar with the oxyplot. Do you have any user manual for me reference? Very Appreciate your help.

**Comment by VisualMelon** [comment]
There is documentation, but it's a bit sparse. https://oxyplot.readthedocs.io/en/latest/ is the nice presentation of it, you can find more up-to-date stuff in the [docs repo](https://github.com/oxyplot/docs/blob/master/models). The best thing to do is to take a look at the examples in the example library, to get a feel for how things can be done, and take a look through the methods in `PlotModel` and any series/annotations you use.
